core/src/main/java/com/google/cloud/sql/core/CoreSocketFactory.java

```diff
@@ -167,16 +148,46 @@ static ListeningScheduledExecutorService getDefaultExecutor() {
         MoreExecutors.getExitingScheduledExecutorService(executor));
   }
 
+  /** Extracts the Unix socket argument from specified properties object. If unset, returns null. */
+  private static String getUnixSocketArg(Properties props) {
+    String unixSocketPath = props.getProperty(UNIX_SOCKET_PROPERTY);
+    if (unixSocketPath != null) {
+      // Get the Unix socket file path from the properties object
+      return unixSocketPath;
+    } else if (System.getenv("CLOUD_SQL_FORCE_UNIX_SOCKET") != null) {
+      // If the deprecated env var is set, warn and use `/cloudsql/INSTANCE_CONNECTION_NAME`
+      // A socket factory is provided at this path for GAE, GCF, and Cloud Run
+      logger.warning(
+          String.format(
+              "\"CLOUD_SQL_FORCE_UNIX_SOCKET\" env var has been deprecated. Please use"
+                  + " '%s=\"/cloudsql/INSTANCE_CONNECTION_NAME\"' property in your JDBC url"
```

Comment:
I don't understand the reference to a URL here. Aren't you asking them to set a property?

Reply:
The JDBC specification uses a url to represent database connections, for example:

`jdbc:postgresql:///<DATABASE_NAME>?unixSocketPath=</PATH/TO/UNIX/SOCKET>&cloudSqlInstance=<INSTANCE_CONNECTION_NAME>`

The driver parses it into a bunch of objects that are passed into the ConnectionFactory, including the Properties object from the query string.

Reply:
Got it.